exams/views.py

```diff
@@ -10,3 +18,27 @@ class PearsonCallbackRedirectView(RedirectView):
     """
     def get_redirect_url(self, status):  # pylint: disable=arguments-differ
         return "/dashboard?exam={status}".format(status=status)
+
+
+class PearsonSSO(APIView):
+    """
+    Views for the Pearson SSO API
+    """
+    authentication_classes = (
+        authentication.SessionAuthentication,
+        authentication.TokenAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated, )
+
+    def post(self, request, *args, **kargs):  # pylint: disable=unused-argument, no-self-use
+        """
+        handle the post request
+        """
+        # import ipdb
+        # ipdb.set_trace()
+        student_id = request.data.get('student_id')
```

Comment:
None of these values should pull from the query, we can't trust user input and this could allow someone to authenticate as another user. Instead, you should pull the student id from the current user's profile. Timestamp can be `datetime.now(pytz.utc)` and `session_timeout` I believe I described on the original issue how to get.

Also, this method should check that there is at least an `ExamProfile` record for the user in a success state. If not it should return an error that the ui can display saying they're not ready for an exam yet (similar messaging to the same state in the exam card is fine I think).